test/integration_test/migration_features_test.go

```diff
@@ -0,0 +1,336 @@
+
+	logrus.Infof("Delete Destination and Source Ns")
+	err = core.Instance().DeleteNamespace(appData.Ns)
+	if err != nil {
+		logrus.Errorf("Error deleting namespace %s: %v\n", appData.Ns, err)
+	}
+	err = setSourceKubeConfig()
+	require.NoError(t, err, "Error setting source kubeconfig")
+
+	err = asyncdr.DeleteAndWaitForMigrationDeletion(mig.Name, mig.Namespace)
+	require.NoError(t, err, "Error deleting migration")
+
+	asyncdr.DeleteCRAndUninstallCRD(appData.OperatorName, appPath, appData.Ns)
+	err = core.Instance().DeleteNamespace(appData.Ns)
+	if err != nil {
+		logrus.Errorf("Error deleting namespace %s: %v\n", appData.Ns, err)
+	}
+	logrus.Infof("Test %s ended", t.Name())
+}
+
+func updateAppReg(appName string, suspendOptions storkapi.SuspendOptions, stashStrategy storkapi.StashStrategy) error {
+	operatorCRMap := getSupportedOperatorCRMapping()
+	if _, ok := operatorCRMap[appName]; !ok {
+		return fmt.Errorf("app %s is not currently supported in test framework", appName)
+	}
+	appResources := operatorCRMap[appName]
+
+	for _, appResource := range appResources {
+		appregName := strings.ToLower(appResource.Kind)
+		reg, err := storkops.Instance().GetApplicationRegistration(appregName)
+		if err != nil {
+			return fmt.Errorf("error getting application registration %s: %w", appregName, err)
+		}
+		for i, resource := range reg.Resources {
+			resource.SuspendOptions = suspendOptions
+			resource.StashStrategy = stashStrategy
+			reg.Resources[i] = resource
+		}
+		_, err = storkops.Instance().UpdateApplicationRegistration(reg)
+		if err != nil {
+			return fmt.Errorf("error setting stash strategy in application registration %s, %w", appregName, err)
+		}
+	}
+	return nil
+}
+
+func validateCR(appName string, namespace string, kubeConfig string) (bool, error) {
+	validated := true
+	operatorCRMap := getSupportedOperatorCRMapping()
+	if _, ok := operatorCRMap[appName]; !ok {
+		return validated, fmt.Errorf("app %s is not currently supported in test framework", appName)
+	}
+	resources := operatorCRMap[appName]
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfig)
+	if err != nil {
+		return validated, fmt.Errorf("error building config from kubeconfig failed: %w", err)
+	}
+	resourceCollector := resourcecollector.ResourceCollector{
+		Driver: nil,
+	}
+	err = resourceCollector.Init(config)
+	if err != nil {
+		return validated, fmt.Errorf("error initing resourcecollector while validating CR: %w", err)
+	}
+
+	for _, resource := range resources {
+		objects, _, err := resourceCollector.GetResourcesForType(
+			resource,
+			nil,
+			[]string{namespace},
+			nil,
+			nil,
+			nil,
+			false,
+			resourcecollector.Options{},
+		)
+		if err != nil {
+			return validated, fmt.Errorf("error fetching objects while validating CR: %w", err)
+		}
+		logrus.Infof("number of resources for app %s in namespace %s is %d", appName, namespace, len(objects.Items))
```

Comment:
We should log GVK here for the resource. Wdyt?

Reply:
the check expects the number of CRs to be present is 1. 
In most cases, if migration has not applied the CR or it has failed, the count will be 0 (no need to print GVK in that case). And for count 1 , its validated. 

Anyway the app's related CR info is already predefined in the test, so debugging is not difficult also.
So I think, we can skip printing GVK for each object in this line.